src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java

```diff
@@ -547,6 +548,21 @@ else if (hand == EquipmentSlot.OFF_HAND)
 		}
 		//CreatureSpawnEvent
 		EventValues.registerEventValue(CreatureSpawnEvent.class, SpawnReason.class, CreatureSpawnEvent::getSpawnReason);
+		//PlayerRespawnEvent - 1.21.5+ added AbstractRespawnEvent as a base class, where prior to that, getRespawnReason was in PlayerRespawnEvent
+		if (Skript.classExists("org.bukkit.event.player.AbstractRespawnEvent")) {
+			EventValues.registerEventValue(PlayerRespawnEvent.class, RespawnReason.class, PlayerRespawnEvent::getRespawnReason);
+		} else {
+			try {
+				var method = PlayerRespawnEvent.class.getMethod("getRespawnReason");
```

Comment:
No need to use var here.
```suggestion
				Method method = PlayerRespawnEvent.class.getMethod("getRespawnReason");
```